### ISSUE TITLE
Filter criteria checking for NOT NULL cannot be added before other criteria

### ISSUE THREAD

**Issue by eriklindgren** [balderdashy/sails-postgresql#159]
**Example**

`.find({ paid_at: { '!': null }, owner: 1 })`

is translated to this database query:
`WHERE "paid_at" IS NOT NULL  AND "owner" = $2`

which results in `ERROR:  could not determine data type of parameter $1` because the NOT NULL criteria takes the place of the first parameter in the query.

A workaround is to always add NOT NULL criteria last.

**Comment by devinivy** [comment]
This is mostly likely a bug in waterline-sequel.  The SQL generated for not-null criteria comes from here: https://github.com/balderdashy/waterline-sequel/blob/cf9ff313faa438cf4601b156f1b888598c63cc2c/sequel/lib/criteriaProcessor.js#L654

**Comment by eriklindgren** [comment]
Yep, looks like `this.paramCount` is always increased at the end of that function.

**Comment by sailsbot** [comment]
Thanks for posting, @eriklindgren.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
